components/eventbrite/sources/new-event-ended/new-event-ended.js

```diff
@@ -0,0 +1,59 @@
+const common = require("../common.js");
+
+module.exports = {
+  ...common,
+  key: "eventbrite-new-event-ended",
+  name: "New Event Ended",
+  description: "Emits an event when an event has ended",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+  hooks: {
+    async deploy() {
+      const params = {
+        orgId: this.organization,
+        params: {
+          status: "ended,completed",
+        },
+      };
+      const events = await this.paginate(
+        this.eventbrite.listEvents.bind(this),
+        "events",
+        params
+      );
+      for (const event of events) {
+        this.emitEvent(event);
+      }
+    },
+  },
+  methods: {
+    ...common.methods,
+    generateMeta(event) {
+      return this.generateEventMeta(event);
```

Comment:
We could have another parent method specific for "events" event sources, with a hierarchy like this:

`common/base <- common/event <- new-event-ended`

What do you think?

Reply:
I added events.js, but I wasn't sure the best way to implement this with new-event-ended using polling and new-event using a webhook. The way I have it set up now works, but seems a bit awkward to me. Any suggestions?

Reply:
Yeah, it seems a bit awkward to me too, especially since the timer-based event source creates an HTTP endpoint that it doesn't use.
The straightforward alternative here is to extend both base components (e.g. `common/event.js` and `common/webhook.js`) in the event source:

```js
const timerBase = require("../common/timer");
const eventBase = require("../common/event");

module.exports = {
  ...timerBase,
  ...eventBase,
  hooks: {
    ...eventBase.hooks,
    async deploy() {
      // Do something specific here...
    },
  },
  methods: {
    ...timerBase.methods,
    ...eventBase.methods,
  },
  async run(event) {
    // Do something specific here...
  },
};
```

Also, `common/event.js` should not extend either `common/webhook.js` or `common/timer.js` since event sources for these events can be either webhook-based or timer-based.

As you can see, it gets complicated really quickly, especially considering that this resembles a multiple inheritance architecture where you might have multiple definitions of the same method/prop/etc. So if your current approach works then let's keep it as is and we can discuss it further and revisit it as part of a future refactor.